docs/index.md

```diff
@@ -113,6 +113,151 @@ expval = pyq.expectation(circ, state, {"theta": theta}, obs)
+            + p_x X \rho X^{\dagger}
+            + p_y Y \rho Y^{\dagger}
+            + p_z Z \rho Z^{\dagger}
+    $$
+* The amplitude damping channel defined as:
+    $$
+        \textbf{AmplitudeDamping}(\rho) =  K_0 \rho K_0^{\dagger} + K_1 \rho K_1^{\dagger}
+    $$
+    with:
+    $\begin{equation*}
+    K_{0} \ =\begin{pmatrix}
+    1 & 0\\
+    0 & \sqrt{1-\ \gamma }
+    \end{pmatrix} ,\ K_{1} \ =\begin{pmatrix}
+    0 & \sqrt{\ \gamma }\\
+    0 & 0
+    \end{pmatrix}
+    \end{equation*}$
+* The phase damping channel defined as:
+    $$
+        \textbf{PhaseDamping}(\rho) = K_0 \rho K_0^{\dagger} + K_1 \rho K_1^{\dagger}
+    $$
+    with:
+    $\begin{equation*}
+    K_{0} \ =\begin{pmatrix}
+    1 & 0\\
+    0 & \sqrt{1-\ \gamma }
+    \end{pmatrix}, \ K_{1} \ =\begin{pmatrix}
+    0 & 0\\
+    0 & \sqrt{\ \gamma }
+    \end{pmatrix}
+    \end{equation*}$
+* The generalize amplitude damping channel is defined as:
+    $$
+        \textbf{GeneralizedAmplitudeDamping}(\rho) = K_0 \rho K_0^{\dagger} + K_1 \rho K_1^{\dagger}
+            + K_2 \rho K_2^{\dagger} + K_3 \rho K_3^{\dagger}
+    $$
+    with:
+$\begin{cases}
+K_{0} \ =\sqrt{p} \ \begin{pmatrix}
+1 & 0\\
+0 & \sqrt{1-\ \gamma }
+\end{pmatrix} ,\ K_{1} \ =\sqrt{p} \ \begin{pmatrix}
+0 & 0\\
+0 & \sqrt{\ \gamma }
+\end{pmatrix} \\
+K_{2} \ =\sqrt{1\ -p} \ \begin{pmatrix}
+\sqrt{1-\ \gamma } & 0\\
+0 & 1
+\end{pmatrix} ,\ K_{3} \ =\sqrt{1-p} \ \begin{pmatrix}
+0 & 0\\
+\sqrt{\ \gamma } & 0
+\end{pmatrix}
+\end{cases}$
+
+ The initialization of noise gates is done as primitives, but they also consider the `probability` that noise affects the system as a parameter. And either a vector or a density matrix can be used as an input, but the output will always be a density matrix.
+
+```python exec="on" source="material-block"
+import torch
+from pyqtorch.noise import AmplitudeDamping, PhaseFlip
+from pyqtorch.utils import random_state
+
+input_state = random_state(n_qubits=2)
+noise_prob = 0.3
+AmpD = AmplitudeDamping(0,noise_prob)
+output_state = AmpD(input_state) #It's a density matrix
+pf = PhaseFlip(1,0.7)
+output_state = pf(output_state)
+```
+
+About the noisy circuit, the initialization is the same as the non-noisy ones and the output will always be a density matrix. Let’s show its used through the simulation of a realistic $X$ gate.
+
+We know that a $X$ gate flips the state of the qubit, for instance $X|0\rangle = |1\rangle$. However, in practice, it's common for the target qubit to stay in its original state after applying $X$ due to the interactions between it and its environment. The possibility of failure can be represented by a Bitflip gate, which flips the state again after the application of the $X$ gate, returning it to its original state with a probability of `1 - gate_fidelity`.
+
+```python exec="on" source="material-block"
+import torch
+import pyqtorch as pyq
+
+import matplotlib.pyplot as plt
+
+input_state = pyq.product_state('00')
```

Comment:
Please import types. It makes it more readable in the end.

Reply:
Done.